### PR TITLE
Create 2023-Q2-Hyperledger-Sawtooth.md

### PR DESCRIPTION
**Pull request by ryjones**

### CODE REVIEW

**Comment by petermetz** [comment]
> Sawtooth is going through a transition from Transact being separated and the idea that Sawtooth 2.0 would use https://github.com/Cargill/splinter .

Meaning that: Transact will be maintained as part of Sawtooth 1.x while Sawtooth 2.x will not be using Transact? (just to make sure I'm reading it correctly)

**Review by petermetz — APPROVED** [review]

**Review by tkuhrt — APPROVED** [review]

**Review by jimthematrix — APPROVED** [review]

**Review by swcurran — APPROVED** [review]

File: project-reports/2023/2023-Q2-Hyperledger-Sawtooth.md
Comment (arsulegai) [inline project-reports/2023/2023-Q2-Hyperledger-Sawtooth.md:10]: [nitpick] The advantage ... that can `[adopt to a wide range of]` consensus mechanisms easily for different workloads as the situation demands.

File: project-reports/2023/2023-Q2-Hyperledger-Sawtooth.md
Comment (arsulegai) [inline project-reports/2023/2023-Q2-Hyperledger-Sawtooth.md:32]: [nitpick] `C`ontinue

File: project-reports/2023/2023-Q2-Hyperledger-Sawtooth.md
Comment (arsulegai) [inline project-reports/2023/2023-Q2-Hyperledger-Sawtooth.md:31]: What is the plan for Sawtooth 2.0? If it new development, do you think there's opportunity for Hyperledger Foundation to help bring in new developers?

**Comment by lehors** [comment]
Why was this report submitted by @ryjones ?

**Comment by lehors** [comment]
It looks to me like the list of maintainers is in need of a serious scrub. Several people listed as maintainers haven't even had a single interaction with github in over a year! Am I missing something here??

File: project-reports/2023/2023-Q2-Hyperledger-Sawtooth.md
Comment (denyeart) [inline project-reports/2023/2023-Q2-Hyperledger-Sawtooth.md:10]: To be honest I'm having some trouble understanding this paragraph and relating it to the Current Plans section. They seem to be talking about different things. Can you clarify what is happening with Transact, Splinter, and each of the releases?

**Comment by ryjones** [comment]
@hyperledger/sawtooth-core-committers @hyperledger/sawtooth-next-committers

File: project-reports/2023/2023-Q2-Hyperledger-Sawtooth.md
Comment (scealiontach) [reply project-reports/2023/2023-Q2-Hyperledger-Sawtooth.md:31]: I believe the point is that Sawtooth 2.x train did not attract new development, meanwhile the 1.x versions had quite a lot of work built on top of it and continuing.  I would also argue that the foundation bringing in help to go counter to the direction that the community is taking is somewhat worse than counter-productive.   We are looking for help, but not for help on paths we are not taking.

File: project-reports/2023/2023-Q2-Hyperledger-Sawtooth.md
Comment (scealiontach) [reply project-reports/2023/2023-Q2-Hyperledger-Sawtooth.md:10]: While we are not going forward with the more drastic redesign that was intended for 2.x, there was nevertheless some good rust work that was done which is not dependent on the full architecture.  Notably, some of this code made its way into Transact and is relatively straightforward to port (in theory).

I cannot comment on releases of Transact or Splinter projects, other than to remind that they each were originally closely related to sawtooth.

File: project-reports/2023/2023-Q2-Hyperledger-Sawtooth.md
Comment (arsulegai) [reply project-reports/2023/2023-Q2-Hyperledger-Sawtooth.md:31]: Thanks @scealiontach for addressing this point. At the beginning of this report I saw mentions of Sawtooth 2.0 but then there were no mentions of it in the plan. For the 1.x version effort, there are ways to engage the community such as meetup events, working sessions, SIG presentations if any work can attract that forum. Looking forward to seeing the project grow. I did not understand later parts of the comment.

File: project-reports/2023/2023-Q2-Hyperledger-Sawtooth.md
Comment (VRamakrishna) [reply project-reports/2023/2023-Q2-Hyperledger-Sawtooth.md:10]: "easily using different consensus mechanisms easily" --> "easily use different consensus mechanisms"

File: project-reports/2023/2023-Q2-Hyperledger-Sawtooth.md
Comment (VRamakrishna) [inline project-reports/2023/2023-Q2-Hyperledger-Sawtooth.md:24]: "answers have been answered" --> "questions have been answered"

File: project-reports/2023/2023-Q2-Hyperledger-Sawtooth.md
Comment (VRamakrishna) [inline project-reports/2023/2023-Q2-Hyperledger-Sawtooth.md:46]: I think you can remove "seamlessly in the Sawtooth project". Rest of the sentence speaks for itself.

**Review by VRamakrishna — APPROVED** [review]

**Review by denyeart — APPROVED** [review]

**Comment by ryjones** [comment]
I converted this to draft to prevent merging.

**Review by mbrandenburger — APPROVED** [review]

File: project-reports/2023/2023-Q2-Hyperledger-Sawtooth.md
Comment (vaporos) [reply project-reports/2023/2023-Q2-Hyperledger-Sawtooth.md:10]: I made the suggestion to possibly rename the '1-3' branch to 'main' based on the idea that if 1.3.x became the focus of new development (based on discussions), it makes sense to have a unified direction. None of those contributions has actually occurred, so there is no current reason to rename the branch or think it will be more active than the current main branch.

Transact is in maintenance mode (project status has changed), but has functionally been merged into Sawtooth lib.

Splilnter's roadmap is here: https://splinter.dev/community/planning/roadmap.html, though development is slow.

**Comment by vaporos** [comment]
> Why was this report submitted by @ryjones ?

Unlike past reports, this report was written by a non-maintainer volunteer in the community, because we are trying very hard to involve active community members in practical ways. My impression is that submitting to github was a barrier, which wasn't something we really anticipated, though I haven't heard this directly from the author.

**Review by bobbi28 — APPROVED** [review]

**Comment by vaporos** [comment]
> It looks to me like the list of maintainers is in need of a serious scrub. Several people listed as maintainers haven't even had a single interaction with github in over a year! Am I missing something here??

This is primarily because the current project policy for maintainers is to not remove them without their permission, out of respect for their contributions to the project and to make it very easy for them to return to contributing to the project. This has had the unintended consequence that in order to remove maintainers, we need to explicitly ask them. We are writing an updated policy that clarifies the criteria for adding maintainers and we will address this in that version.

File: project-reports/2023/2023-Q2-Hyperledger-Sawtooth.md
Comment (vaporos) [reply project-reports/2023/2023-Q2-Hyperledger-Sawtooth.md:31]: To clarify, "I believe the point is that Sawtooth 2.x train did not attract new development" should not have been the take-away from the discussion, but rather that if 1.3.x development exceeded currently main branch development, we can change course and get to 2.x via 1.3.x instead. Its a longer path but would be perfectly fine. "meanwhile the 1.x versions had quite a lot of work built on top of it" -- this is not true currently based strictly on contributions, but I think it would be fair to say instead that there is a lot of interest in continuing 1.3.x development theoretically. I'd rather see a single line of development than two, so if substantial/sustained 1.3.x contributions do materialize, then we need make sure our approach is unified so we are working in the same place (thus my proposal).

@arsulegai There is plenty of areas that new developers could contribute, the easiest of which is with the Rust re-write, which means taking existing python and finish re-writing it into Rust. Or removing or updating the dockerfiles so they work with a newer Ubuntu.

**Comment by vaporos** [comment]
> > Sawtooth is going through a transition from Transact being separated and the idea that Sawtooth 2.0 would use https://github.com/Cargill/splinter .
> 
> Meaning that: Transact will be maintained as part of Sawtooth 1.x while Sawtooth 2.x will not be using Transact? (just to make sure I'm reading it correctly)

We moved Transact into Sawtooth Library crate, which can be used by both current main and 1-3 branches (the library crate has its own component version separate from Sawtooth Core).

**Comment by denyeart** [comment]
Thank you for addressing the comments and adding the clarifications. Could you report itself be updated to reflect these clarifications?

File: project-reports/2023/2023-Q2-Hyperledger-Sawtooth.md
Comment (jimthematrix) [reply project-reports/2023/2023-Q2-Hyperledger-Sawtooth.md:10]: I'm also having a lot of trouble understanding this paragraph. Based on the clarifications above and elsewhere, I suggest the paragraph needs a re-write to describe the different threads:
- Transact has been successfully merged into Sawtooth, due to the project going into the Dormant mode. But there are still contributions to the Transact code base that need to be ported into Sawtooth continuously
- The 1.x branches are the current focus of development activities, and expected to be for the foreseeable future.
- Sawtooth 2.0 is a redesign and will be based on Splinter, which is a component from outside of Hyperledger, but was created with Sawtooth support in mind. At the moment the community's interest for the 2.0 development is low.
- A proposal was made to consider making the 1.3.x branch the new main, to replace the 2.0 work that the main was intended for, to reflect the focus of the community

I hope I was able to capture the various points throughout the discussions, but I'm sure I may have misinterpreted something entirely

**Review by jimthematrix — CHANGES_REQUESTED** [review]
see comment https://github.com/hyperledger/toc/pull/128#discussion_r1269564288

**Comment by ryjones** [comment]
@vaporos please address @jimthematrix 's comment

**Comment by lehors** [comment]
> > It looks to me like the list of maintainers is in need of a serious scrub. Several people listed as maintainers haven't even had a single interaction with github in over a year! Am I missing something here??
> 
> This is primarily because the current project policy for maintainers is to not remove them without their permission, out of respect for their contributions to the project and to make it very easy for them to return to contributing to the project. This has had the unintended consequence that in order to remove maintainers, we need to explicitly ask them. We are writing an updated policy that clarifies the criteria for adding maintainers and we will address this in that version.

Thanks for the explanation. This policy appears to be flawed though because you can find yourself cornered in case people no longer respond and the whole idea of being a *maintainer* is to be maintaining. Listing people as maintainers when they no longer do any maintenance defeats the point.

A better way to acknowledge past contributions is to list them as "Retired maintainers". This is what's done for instance for Fabric: https://github.com/hyperledger/fabric/blob/main/MAINTAINERS.md

**Comment by agunde406** [comment]
> > > It looks to me like the list of maintainers is in need of a serious scrub. Several people listed as maintainers haven't even had a single interaction with github in over a year! Am I missing something here??
> > 
> > 
> > This is primarily because the current project policy for maintainers is to not remove them without their permission, out of respect for their contributions to the project and to make it very easy for them to return to contributing to the project. This has had the unintended consequence that in order to remove maintainers, we need to explicitly ask them. We are writing an updated policy that clarifies the criteria for adding maintainers and we will address this in that version.
> 
> Thanks for the explanation. This policy appears to be flawed though because you can find yourself cornered in case people no longer respond and the whole idea of being a _maintainer_ is to be maintaining. Listing people as maintainers when they no longer do any maintenance defeats the point.
> 
> A better way to acknowledge past contributions is to list them as "Retired maintainers". This is what's done for instance for Fabric: https://github.com/hyperledger/fabric/blob/main/MAINTAINERS.md

As Shawn mentioned, we have been working on an update for our maintainer rules which is currently up in a PR here https://github.com/hyperledger/sawtooth-docs/pull/208. Once it is merged we are going to move the remaining inactive maintainers to Emeritus status.

**Pull request merged**
